src/hotspot/share/gc/shenandoah/shenandoahConcurrentMark.cpp

```diff
@@ -178,7 +178,8 @@ template<ShenandoahGenerationType GENERATION>
 void ShenandoahMarkConcurrentRootsTask<GENERATION>::work(uint worker_id) {
   ShenandoahConcurrentWorkerSession worker_session(worker_id);
   ShenandoahObjToScanQueue* q = _queue_set->queue(worker_id);
-  ShenandoahObjToScanQueue* old = _old_queue_set == nullptr ? nullptr : _old_queue_set->queue(worker_id);
+  ShenandoahObjToScanQueue* old = (_old_queue_set == nullptr) ?
```

Comment:
call this `old_q` too like you did above at line 121?

Reply:
Done.